docs/services/compute.rst

```diff
@@ -9,11 +9,29 @@ Globus Compute
     :show-inheritance:
     :exclude-members: error_class, scopes
 
+..  autoclass:: ComputeClientV2
+    :members:
+    :member-order: bysource
+    :show-inheritance:
+    :exclude-members: error_class, scopes
+
+    ..  attribute:: scopes
+
+        ..  listknownscopes:: globus_sdk.scopes.ComputeScopes
+            :base_name: ComputeClient.scopes
+
+..  autoclass:: ComputeClientV3
+    :members:
+    :member-order: bysource
+    :show-inheritance:
+    :exclude-members: error_class, scopes
+
```

Comment:
Each client's docstring indicates the Globus Compute API version that it supports, highlighting the difference between the clients. The choice of one client over another depends on the users specific needs or preferences, which could be different for each route.

Did you have something specific in mind to add?

Reply:
This came up when I raised this topic earlier today and I quickly agreed that now that we're giving users a menu of options, we need to give them some kind of narrative or meta explanation which enables them to navigate those options.

We've given users a number of similar classes here, which they may want to use to access the service's capabilities. What should a user looking to script against Compute do with these clients?

IMO it's okay if we want to tell users that they should be looking at the Compute SDK instead, or, instead, we could say "prefer the V3 client." But in any case we need some explanation of what these classes are and how a user should interact with them.

I'm also not at all averse to advertising some deprecation timeline for the V2 APIs, if that is the plan. We can make corrections to the docs if they drift from reality, but it would allow users to grasp that V2 is current, V3 is new, and that building on top of the V2 client exposes them to a pending deprecation -- therefore, they may prefer to wait for V3 support rather than implementing solutions under V2.

Reply:
I just pushed an update that points standard users to the Globus Compute SDK (nice suggestion!) and has some additional information on the clients.

Re: deprecation of V2, we currently plan to support V2 until Globus SDK V4, so I don't think a deprecation timeline is necessary.

Reply:
I'm in favor of the new text. I think it appropriately captures that these are an underlying implementation and that you may use them if you wish, but that you're opting for a "closer to the metal" experience, rather than having some potentially nice higher-level helpers.
I also don't think the text is set in stone -- we can always revise and refine it if we later decide that it sends too many users through the Compute SDK when their needs can be met with `globus-sdk` components.

---

Very minor note: should we be linking to `latest` docs or `stable`? I assume the RTD config is similar to what we have for `globus-sdk` where `stable` is the last release and `latest` is `main`. If so, I think linking to `stable` is probably better.

Reply:
Yep, that docs update is plenty.

We don't need to describe all of the differences between the clients, but we should provide some degree of instruction for a newcomer (including linking out to more authoritative compute docs). I'm happy with what you've added. Thanks

Reply:
OK, great. I just pushed a commit to change the link from `latest` to `stable`. Otherwise, we should be all set.